frontend/src/services/backend/authentication_backend.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Communicates with the backend to exchange an auth code and a redirect 
+ * URI for an ID token.
+ */
+export class AuthenticationBackendService {
+  /** Returns the ID token given an auth code and redirect URI. */
+  async retrieveToken(code: string, redirectUri: string): Promise<string> {
+    let bodyContents = {
+      code: code,
+      redirectUri: redirectUri,
+    }
+
+    return fetch("/api/retrieve-token", { 
+      method: "POST",
+      body: JSON.stringify(bodyContents),
+    }).then(response => response.text());
```

Comment:
nit: Since this is async, you should prefer `await`ing `fetch` over using `.then`. Both work but you should try to avoid mixing them where possible.

Reply:
Good point! I've switched .then to an await and pushed the fix.